masonry/src/passes/accessibility.rs

```diff
@@ -99,20 +100,28 @@ fn build_access_node(widget: &dyn Widget, state: &WidgetState, scale_factor: f64
             .collect::<Vec<NodeId>>(),
     );
 
-    if state.is_hot {
-        node.set_hovered();
+    node
+}
+
+fn set_common_properties(ctx: &mut AccessCtx) {
+    if ctx.is_hot() {
+        ctx.current_node().set_hovered();
     }
-    if state.is_disabled {
-        node.set_disabled();
+    if ctx.is_disabled() {
+        ctx.current_node().set_disabled();
     }
-    if state.is_stashed {
-        node.set_hidden();
+    if ctx.is_stashed() {
+        ctx.current_node().set_hidden();
     }
-    if state.clip.is_some() {
-        node.set_clips_children();
+    if ctx.widget_state.clip.is_some() {
+        ctx.current_node().set_clips_children();
+    }
+    if ctx.is_in_focus_chain() && !ctx.is_disabled() {
+        ctx.current_node().add_action(accesskit::Action::Focus);
+    }
+    if ctx.is_focused() {
+        ctx.current_node().add_action(accesskit::Action::Blur);
     }
```

Comment:
Is there a reason this is split out from `build_access_node`?

Reply:
... https://github.com/linebender/xilem/pull/596#discussion_r1765187247

Essentially, build_access_node doesn't have the AccessCtx, which means it would either need to duplicate the logic of the context methods or not have this functionality implemented properly.